library/std/src/sys/sgx/abi/usercalls/tests.rs

```diff
@@ -17,12 +17,12 @@ fn test_copy_function() {
             dst.copy_from_enclave(&[0u8; 100]);
 
             // Copy src[0..size] to dst + offset
-            unsafe { copy_to_userspace(src.as_ptr(), dst.as_mut_ptr().offset(offset), size) };
+            unsafe { copy_to_userspace(src.as_ptr(), dst.as_mut_ptr().add(offset), size) };
 
             // Verify copy
             for byte in 0..size {
                 unsafe {
-                    assert_eq!(*dst.as_ptr().offset(offset + byte as isize), src[byte as usize]);
+                    assert_eq!(*dst.as_ptr().add(offset + byte), src[byte as usize]);
```

Comment:
sneaky: `offset` was `{integer}` previously inferred to `isize`, now to `usize`.

Reply:
Looks like `byte` was previously `i32` (because unconstrained, as it was always `as`ed) and is now `usize`?  So it can probably be `src[byte]` too.

(It would be nice to have a warning for T-to-T `as` casts...)

Reply:
I was thinking about linting T2T casts too, while doing these refactorings. It should probably be easy to add 🤔

Reply:
Hm, there is a clippy lint for that ([docs](https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast), [source](https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/casts/unnecessary_cast.rs)), should I open an uplifting issue/pr?

Reply:
I guess the problem is if you want to keep it -- say it's there to be resilient to `c_int` widths or something, where it's `c_int as i64`, which might be a nop sometimes.

For a clippy lint saying "hey, just allow it" is fine, but for rustc lints we generally want to be able to give a "or write \_\_\_\_\_\_ instead to make it obvious why you want to keep it like this" suggestion.

Nothing jumped to mind to me for how to do that, but hopefully you'll come up with something clever to lint only in the valuable places.

Reply:
Not linting `i64 as c_int` is probably possible by recording in hir (?) that `c_int` is a type alias, but not linting `c_int as i64` I think it impossible, there is simply no way to differentiate a variable of type `c_int` with a variable of type `i64` I believe (if `c_int` is an alias to `i64`)

Reply:
As a second thought, would this still be useful as a allow-by-default lint? 🤔